test/e2e/specs/all-features-confirmation/recordset/presentation.spec.ts

```diff
@@ -0,0 +1,804 @@
+
+  // NOTE: should this NOT run in CI still?
+  test('For recordset with columns with waitfor', async ({ page, baseURL }, testInfo) => {
+    const params = testParams.active_list;
+    const data = params.data;
+
+    const PAGE_URL = `/recordset/#${getCatalogID(testInfo.project.name)}/${params.schema_name}:${params.table_name}`;
+
+    await test.step('should load recordset page', async () => {
+      await page.goto(`${baseURL}${PAGE_URL}@sort(${params.sortby})`);
+      await RecordsetLocators.waitForRecordsetPageReady(page);
+
+      // "wait for" table column aggregate spinners to all be hidden by waiting until the count retruned in 0
+      // NOTE: using `waitForAggregates` fails because there are "multiple spinners on the page"
+      await expect.soft(RecordsetLocators.getAggregateSpinners(page)).toHaveCount(0);
+    });
+
+    await test.step('should not show the total count if hide_row_count is true.', async () => {
+      await testTotalCount(page, `Displaying all${data.length}records`)
+    });
+
+    await test.step('should show correct table rows.', async () => {
+      await testRecordsetTableRowValues(page, data, true);
+    });
+
+    await test.step('going to a recordset page with no results, the loader for columns should hide.', async () => {
+      await page.goto(`${baseURL}${PAGE_URL}/main_id=03`);
+      await RecordsetLocators.waitForRecordsetPageReady(page);
+      await RecordsetLocators.waitForAggregates(page);
+    })
+  });
+
+  test(`For table ${testParams.accommodation_tuple.table_name}`, async ({ page, baseURL }, testInfo) => {
+    const params = testParams.accommodation_tuple;
+    const catalogID = getCatalogID(testInfo.project.name);
+    const PAGE_URL = `/recordset/#${catalogID}/${params.schema_name}:${params.table_name}`
+
+    await test.step('should load recordset page', async () => {
+      await page.goto(`${baseURL}${PAGE_URL}/${params.key}@sort(${params.sortby})`);
+      await RecordsetLocators.waitForRecordsetPageReady(page);
+
+      // NOTE: same comment from above about multiple spinners on the page
+      await expect.soft(RecordsetLocators.getAggregateSpinners(page)).toHaveCount(0);
+    });
+
+    await test.step('presentation of the recordset page', async () => {
+      if (!process.env.CI) {
+        await test.step('delete files', async () => {
+          // delete files that may have been downloaded before
+          await deleteDownloadedFiles(params.file_names.map((name: string) => {
+            return `${DOWNLOAD_FOLDER}/${name}`
+          }));
+        });
+      }
+
+      await test.step(`should have ${params.title} as title`, async () => {
+        await expect.soft(RecordsetLocators.getPageTitleElement(page)).toHaveText(params.title);
+      });
+
+      await test.step('should have the correct tooltip.', async () => {
+        await testTooltip(RecordsetLocators.getPageTitleElement(page).locator('.chaise-icon-for-tooltip'), params.comment, APP_NAMES.RECORDSET);
```

Comment:
We should avoid calling `.locator()` here. You should create a function for this instead. Similar to `getPageTitleTooltip` in `chaise.page.js`.